ts-client/src/dlmm/types/index.ts

```diff
@@ -188,6 +189,57 @@ export interface BinLiquidity {
   pricePerToken: string;
 }
 
+export module BinLiquidity {
+  export function fromBin(
+    bin: Bin,
+    binId: number,
+    binStep: number,
+    baseTokenDecimal: number,
+    quoteTokenDecimal: number,
+    version?: number,
+  ): BinLiquidity {
+    const pricePerLamport = getPriceOfBinByBinId(
+      binId,
+      binStep
+    ).toString();
+    return {
+      binId,
+      xAmount: bin.amountX,
+      yAmount: bin.amountY,
+      supply: bin.liquiditySupply,
+      price: pricePerLamport,
+      version: version ?? 1,
```

Comment:
we should pass the binArray.version here

Reply:
> we should pass the binArray.version here

version is chosen explicitly: https://github.com/MeteoraAg/dlmm-sdk/blob/4eb8a857945f83826f21b14234388050a08a3904/ts-client/src/dlmm/index.ts#L5325

If there's no any bin array found (assuming it's a case for a new pool) then `version = 1` is used. It mimics the behaviour of the current sdk version.